### PR TITLE
Minor spotbugs fixes in respect to null checks

### PR DESCRIPTION
**Pull request by StefanSpieker**
Minor spotbugs fixes in respect to null checks. And added missing `@NonNull` annotations.

### Proposed changelog entries

* N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by oleg-nenashev — DISMISSED** [review]
`workDirSettings` in `readResolve` is needed for data migration

**Comment by StefanSpieker** [comment]
@oleg-nenashev: What should I do then? 
1. Should I disable the `RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE` with a `@SuppressFBWarnings(value = "RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE", justification = "workDirSettings in readResolve is needed for data migration.")` or 
2. should I set `workDirSettings` to `@CheckForNull`?

**Comment by oleg-nenashev** [comment]
The first option looks good to me

File: core/src/main/java/hudson/tasks/Maven.java
Comment (basil) [inline core/src/main/java/hudson/tasks/Maven.java:244]: Oleg's comment about `readResolve` also applies here. Do not break `readResolve()` functionality. See [Hint on retaining backward compatibility](https://wiki.jenkins.io/display/JENKINS/Hint+on+retaining+backward+compatibility). On a side note, that content really needs to be modernized and moved to the [Jenkins Developer Reference](https://www.jenkins.io/doc/developer/book/) on jenkins.io. Volunteers would be appreciated, but this is not related to your change. The only thing necessary for this PR is to preserve existing `readResolve()` functionality.

Comment (StefanSpieker) [reply core/src/main/java/hudson/tasks/Maven.java]: Thanks for pointing that out, I'll take a look at the docs. If I find time I will create a PR for that.

**Review by timja — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
